docs/guides/nwaku/configure-nwaku.md

```diff
@@ -0,0 +1,246 @@
+
+```bash
+./build/wakunode2 --storenode=[STORE PEER MULTIADDR]
+```
+
+For example, consider a `nwaku` node that does not persist messages but can query peers for historical messages:
+
+```bash
+./build/wakunode2 --storenode=/dns4/node-01.ac-cn-hongkong-c.wakuv2.prod.statusim.net/tcp/30303/p2p/16Uiu2HAm4v86W3bmT1BiH6oSPzcsSr24iDQpSN5Qa992BCjjwgrD
+```
+
+## Generate and Configure a Node Key
+
+Nodes generate [new random key pairs](/overview/reference/glossary#node-key) at each boot, leading to different `multiaddrs`. To maintain consistency, you can use a pre-generated private key with the `nodekey` option:
+
+```bash
+./build/wakunode2 --nodekey=[NODE PRIVATE KEY]
+```
+
+This option takes a [Secp256k1](https://en.bitcoin.it/wiki/Secp256k1) private key (64-char hex string). On Linux, you can use the OpenSSL `rand` command for a pseudo-random 32-byte hex string:
+
+```bash
+openssl rand -hex 32
+
+# 286cae9f2990bfc49dafdd3a9e737f56ddba3656e5e427108cef456fb67680e8
+```
+
+On Linux, you can create a reusable key file using OpenSSL. To get the 32-byte private key in hex format, use the `ecparam` command and some standard utilities:
+
+```bash
+# Generate key file
+openssl ecparam -genkey -name secp256k1 -out my_private_key.pem
+
+# Extract 32-byte private key
+openssl ec -in my_private_key.pem -outform DER | tail -c +8 | head -c 32| xxd -p -c 32
+
+# read EC key
+# writing EC key
+# 286cae9f2990bfc49dafdd3a9e737f56ddba3656e5e427108cef456fb67680e8
+```
+
+You can use the output `286cae9f2990bfc49dafdd3a9e737f56ddba3656e5e427108cef456fb67680e8` as a `Node Key` for `nwaku`:
+
+```bash
+./build/wakunode2 --nodekey=286cae9f2990bfc49dafdd3a9e737f56ddba3656e5e427108cef456fb67680e8
+```
+
+## Configure WebSocket Transport
+
+WebSocket is the only [transport method](/overview/concepts/transports) browser nodes support using [@waku/sdk](/guides/js-waku/). To enable WebSocket in `nwaku` to serve browser peers, use the following configuration options:
+
+- `websocket-support`: Enables WebSocket (`ws`) on the node (disabled by default).
+- `websocket-port` (optional): WebSocket listening port. If you omit this option, it will default to `8000`.
+- `websocket-secure-support`: Enables Secure WebSocket (`wss`) on the node (disabled by default).
+- `websocket-secure-key-path`: Secure WebSocket key path.
+- `websocket-secure-cert-path`: Secure WebSocket Certificate path.
+
+```bash
+./build/wakunode2 \
+  --websocket-support=true \
+  --websocket-port=[WEBSOCKET LISTENING PORT] \
+  --websocket-secure-support=true \
+  --websocket-secure-key-path=[SECURE WEBSOCKET KEY PATH] \
+  --websocket-secure-cert-path=[SECURE WEBSOCKET CERTIFICATE PATH]
+```
+
+:::info
+Secure WebSocket (`wss`) is not required for local development, as the [secure context restriction](/guides/js-waku/debug-waku-dapp#checking-websocket-setup) does not apply when your website is served locally, like on `localhost` or `127.0.0.1`.
```

Comment:
Feels a bit out of context as nowhere was `wss` specified to be required. Maybe in the intro (L133) specify that _Secure_ WebSocket is the only transport method browser nodes support.